test/thorest-api/transactions/setup/transaction-data-driven-flow.ts

```diff
@@ -24,72 +22,123 @@ class TransactionDataDrivenFlow {
+            )
         }
 
+        const { expectedResult } = this.plan.getTxStep
+
         const tx = await Node1Client.getTransaction(txId, {
             pending: true,
         })
 
         expectedResult(tx)
     }
 
-    private async getTransactionReceipt(txId?: string): Promise<string | undefined> {
+    private async getTransactionReceipt(
+        txId?: string,
+    ): Promise<components['schemas']['ReceiptMeta'] | undefined> {
         if (!this.plan.getTxReceiptStep) {
             return
         }
 
-        const { expectedResult } = this.plan.getTxReceiptStep
         if (!txId) {
-            throw new TestCasePlanStepError('getTransactionReceipt step expected to execute, but txId is not defined')
+            throw new TestCasePlanStepError(
+                'getTransactionReceipt step expected to execute, but txId is not defined',
+            )
         }
 
+        const { expectedResult } = this.plan.getTxReceiptStep
+
         const receipt = await pollReceipt(txId)
 
         expectedResult(receipt)
 
-        return receipt.meta?.blockID
+        return receipt.meta
+    }
+
+    private async getLogTransfer(
+        block: components['schemas']['ReceiptMeta'] | undefined
+    ) {
+        if (!this.plan.getLogTransferStep) {
+            return
+        }
+
+        if (!block) {
+            throw new TestCasePlanStepError(
+                'getLogTransfer step expected to execute, but block is not defined',
+            )
+        }
+
+        const { blockNumber } = block
+        const { expectedResult, logFilters } = this.plan.getLogTransferStep
+        const request = logFilters ?? createDefaultLogFilterRequest(blockNumber)
+
+        const logsResponse = await Node1Client.queryTransferLogs(request)
+
+        expectedResult(logsResponse, block)
     }
 
-    private async getTransactionBlock(blockId: string | undefined, txId: string | undefined) {
+    private async getTransactionBlock(
+        blockId: string | undefined,
+        txId: string | undefined,
+    ) {
         if (!this.plan.getTxBlockStep) {
             return
         }
 
-        const { expectedResult } = this.plan.getTxBlockStep
         if (!blockId || !txId) {
-            throw new TestCasePlanStepError('getTransactionBlock step expected to execute, but txId or blockId are not defined')
+            throw new TestCasePlanStepError(
+                'getTransactionBlock step expected to execute, but txId or blockId are not defined',
+            )
         }
 
+        const { expectedResult } = this.plan.getTxBlockStep
+
         const block = await Node1Client.getBlock(blockId)
 
         expectedResult({ block, txId })
     }
 }
 
-export {
-    TransactionDataDrivenFlow
+function createDefaultLogFilterRequest(
+    blockNumber: number | undefined
+): components['schemas']['TransferLogFilterRequest'] {
+    return {
+        range: {
+            to: blockNumber,
+            from: blockNumber,
```

Comment:
should we have block range or just block - block is enough ?

Reply:
This is something that can be used as a default log filter: it looks for the block the tx is added into, but if you want to have a customized filter you can pass it directly when creating the testPlan in the transaction log step